compat/mingw.c

```diff
@@ -3337,6 +3338,45 @@ static int is_system32_path(const char *path)
 	return 1;
 }
 
+/*
+ * Extract the Windows OS build number from the registry, this is
+ * best method of those available here:
+ * https://stackoverflow.com/a/47583450/262458
+ *
+ * This post explains how to read the registry using WinAPI:
+ * https://stackoverflow.com/a/35717/262458
+ */
+unsigned windows_build(void)
+{
+	static unsigned is_cached    = 0;
+	static unsigned cached_value = 0;
+
+	if (is_cached)
+		return cached_value;
+
+	HKEY hkey;
+	ULONG res = RegOpenKeyExA(HKEY_LOCAL_MACHINE, "SOFTWARE\\Microsoft\\Windows NT\\CurrentVersion", 0, KEY_READ, &hkey);
+
+	if (res != ERROR_SUCCESS) goto err;
+
+	char buf[512];
+	int bufsize = sizeof(buf);
+
+	res = RegQueryValueExA(hkey, "CurrentBuildNumber", 0, NULL, (LPSTR)buf, &bufsize);
```

Comment:
I think we already have a better way to get to this via `GetVersion()` (see e.g. [this existing example](https://github.com/git-for-windows/git/blob/v2.35.1.windows.2/compat/mingw.c#L3706-L3718)).

Reply:
> The `GetVersion()` code did not work for me,

How so? Does it report the wrong version? If so, I suspect that [these lines are not enough anymore](https://github.com/git-for-windows/git/blob/v2.35.1.windows.2/compat/win32/git.manifest#L21-L22).

> and the compiler says it's deprecated:

That may be so, but we're still using it. As long as we do that, we should do it consistently instead of introducing inconsistencies in how we obtain the build number.

Reply:
Here are my results with `GetVersion()` on my system:

![screenshot of GetVersion()](https://i.imgur.com/Lj1dMqu.png)

Reply:
If you like my method, and IMHO it's the most sane one available, I will offer to do a separate PR to use it everywhere.

Reply:
It does in fact work with the manifest.

![manifest screenshot](https://i.imgur.com/H49uQgw.png)

Reply:
I need to check if the VS build actually includes the manifest, because I don't think it does, at least I vaguely remember this not working for me last I tried it.

If that is the case I will open a new PR for this.

Reply:
Indeed, it seems that the `git.manifest` fell between the cracks.